subsys/esb/esb.c

```diff
@@ -1402,9 +1376,33 @@ static void on_radio_disabled_tx_wait_for_ack(void)
 
 		if ((tx_fifo.count == 0) || (esb_cfg.tx_mode == ESB_TXMODE_MANUAL)) {
 			esb_state = ESB_STATE_IDLE;
-			errata216_off();
+
+			if (IS_ENABLED(CONFIG_ESB_KEEP_TX_IDLE)) {
+				nrf_radio_shorts_set(NRF_RADIO, 0);
+				nrf_radio_task_trigger(NRF_RADIO, NRF_RADIO_TASK_TXEN);
+				while (!nrf_radio_event_check(NRF_RADIO, NRF_RADIO_EVENT_READY)) {
+					/* Do nothing */
+				}
+				nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_READY);
+			} else if (fast_switching) {
+				on_radio_event = NULL;
+				nrf_radio_shorts_set(NRF_RADIO, 0);
+				nrf_radio_task_trigger(NRF_RADIO, NRF_RADIO_TASK_DISABLE);
+			}
+
+			if (!IS_ENABLED(CONFIG_ESB_KEEP_TX_IDLE)) {
+				errata216_off();
+			}
+
 			set_evt_interrupt();
 		} else {
+			if (IS_ENABLED(CONFIG_ESB_KEEP_TX_IDLE)) {
+				nrf_radio_task_trigger(NRF_RADIO, NRF_RADIO_TASK_TXEN);
+				while (!nrf_radio_event_check(NRF_RADIO, NRF_RADIO_EVENT_READY)) {
```

Comment:
Please add a comment for clarity why there is a need to block here waiting for the Ready event.

Reply:
This block of code was removed. For the ACK package, the radio goes to the DISABLED state.